modules/videoio/src/cap.cpp

```diff
@@ -117,6 +117,26 @@ bool VideoCapture::open(const String& filename, int apiPreference, const std::ve
 
     const VideoCaptureParameters parameters(params);
     const std::vector<VideoBackendInfo> backends = cv::videoio_registry::getAvailableBackends_CaptureByFilename();
+
+    if(apiPreference != CAP_ANY)
+    {
+        //Check if the backend generally is available but can't be used to capture by filename
+        bool foundFilenameBackend = false;
+        for (size_t i = 0; i < backends.size(); i++)
+        {
+            if((foundFilenameBackend = (apiPreference == backends[i].id)))
+                break;
+        }
```

Comment:
Instead of scanning in multiple places we could add internal call:
`const VideoBackendInfo& cv::videoio_registry::getBackendInfo(VideoCaptureAPIs api)`
(similar to `cv::videoio_registry::getBackendName()` but non-public).

And then use `VideoBackendInfo::mode` field directly.

Reply:
Sounds like a plan

Reply:
I am not sure what internal in that case means. ```cv::videoio_registry::internal```?

Reply:
"internal" means that it should not be a part of OpenCV **public** API.

- declaration should go to `src/videoio_registry.hpp`
- definition/implementation `src/videoio_registry.cpp`

You could add `::internal` nested namespace as your wish.

Reply:
thx!

Reply:
Done